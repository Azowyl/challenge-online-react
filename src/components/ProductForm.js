import React, {Component, useState} from "react";

import "./components.scss";

function ProductForm() {
  const [productData, setProductData] = useState({
    product: "",
    count: "",
    price: "",
    promotionalPrice: "",
  });

  const handleInputChange = (field, event) => {
    return setProductData({[field]: event.target.value})
  }

  const handleSubmit = (event) => {
    event.preventDefault();
  }

  return (
    <div className="centered">
      <h2>Add new product</h2>
      <div>
        <label className="label" htmlFor="">Product name</label>
        <input name="product" type="text" value={ productData.product } onChange={handleInputChange}/>
        <label className="label" htmlFor="">Items count</label>
        <input name="count" type="text" value={ productData.count } onChange={handleInputChange}/>
        <label className="label" htmlFor="">Price</label>
        <input name="price" type="text" value={ productData.price } onChange={handleInputChange}/>
        <label className="label" htmlFor="">Promotional Price</label>
        <input name="promotionalPrice" type="text" value={ productData.promotionalPrice } onChange={handleInputChange}/>
        <input className="button" type="submit" value="Aceptar" />
      </div>
    </div>
  );
}

export default ProductForm;
