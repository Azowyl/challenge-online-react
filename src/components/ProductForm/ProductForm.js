import React, {useState} from "react";
import {useProductContext} from "../Providers/ProductListProvider";
import Input from "../Input/Input";

import "./productForm.scss";
import BottomSpacer from "../Spacing/BottomSpacer";
import SubmitButton from "../Button/SubmitButton";
import {CopyIcon, MoneyIcon, TagIcon, VolumeIcon} from "@tiendanube/icons";

function ProductForm({location}) {
  const [productData, setProductData] = useState(location.state || {
    name: "",
    count: "",
    price: "",
    promotionalPrice: "",
  });
  const [errors, setErrors] = useState({
    name: "",
    count: "",
    price: "",
    promotionalPrice: "",
  });

  const { addProduct } = useProductContext();

  const handleInputChange = (event) => {
    return setProductData({...productData, [event.target.id]: event.target.value})
  }

  const handleSubmit = (event) => {
    event.preventDefault();

    const errors = getFormErrors();
    if (Object.keys(errors).length > 0) {
        setErrors(errors);
    } else {
        addProduct(productData);
    }
  }

  const getFormErrors = () => {
    let errors = {};
    Object.entries(productData).forEach(([key, value]) => {
      if (!value) {
        errors[key] = "This field is required";
      }
    });

    return errors;
  }

  return (
    <div className="centered">
      <h2>Add new product</h2>
      <form onSubmit={handleSubmit}>
        <Input
            name="name"
            type="text"
            value={productData.name}
            onChange={handleInputChange}
            label="Product name"
            error={errors.name}
            icon={<TagIcon />}
        />
        <BottomSpacer />
        <Input
            name="count"
            type="number"
            value={productData.count}
            onChange={handleInputChange}
            label="Items count"
            error={errors.count}
            icon={<CopyIcon />}
            min={0}
        />
        <BottomSpacer />

        <Input
            name="price"
            type="number"
            value={productData.price}
            onChange={handleInputChange}
            label="Price"
            error={errors.price}
            icon={<MoneyIcon />}
            min={0}
        />
        <BottomSpacer />
        <Input
            name="promotionalPrice"
            type="number"
            value={productData.promotionalPrice}
            onChange={handleInputChange}
            label="Promotional Price"
            error={errors.promotionalPrice}
            icon={<VolumeIcon />}
            min={0}
        />
        <BottomSpacer />

        <SubmitButton label="Aceptar" />
      </form>
    </div>
  );
}

export default ProductForm;
