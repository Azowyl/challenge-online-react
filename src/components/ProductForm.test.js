import React from 'react';
import {render} from "@testing-library/react";
import ProductForm from "./ProductForm";

describe('ProductForm', () => {
    let renderComponent;

    beforeEach(() => {
        renderComponent = () => render(<ProductForm />);
    })

    it('renders ProductForm correctly', () => {
        expect(renderComponent().asFragment()).toMatchSnapshot();
    });
})