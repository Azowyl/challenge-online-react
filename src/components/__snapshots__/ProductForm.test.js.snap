// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductForm renders form correctly 1`] = `
<DocumentFragment>
  <div
    class="centered"
  >
    <h2>
      Add new product
    </h2>
    <form>
      <label
        class="label"
        for="name"
      >
        Product name
      </label>
      <input
        id="name"
        name="name"
        type="text"
        value=""
      />
      <label
        class="label"
        for="count"
      >
        Items count
      </label>
      <input
        id="count"
        name="count"
        type="text"
        value=""
      />
      <label
        class="label"
        for="price"
      >
        Price
      </label>
      <input
        id="price"
        name="price"
        type="text"
        value=""
      />
      <label
        class="label"
        for="promotionalPrice"
      >
        Promotional Price
      </label>
      <input
        id="promotionalPrice"
        name="promotionalPrice"
        type="text"
        value=""
      />
      <input
        class="button"
        type="submit"
        value="Aceptar"
      />
    </form>
  </div>
</DocumentFragment>
`;
